bazelrio/scripts/deploy/Deploy.java

```diff
@@ -1,167 +1,240 @@
+        .type(String.class)
+        .help("the executable binary to deploy")
+        .required(true);
+    parser.addArgument("--robot_command")
+        .type(String.class)
+        .help(
+            "the script to run the robot binary (deployed to /home/lvuser/robotCommand)")
+        .setDefault("{}");
+    parser.addArgument("--team_number")
+        .type(Integer.class)
+        .help("the team number, used for the roboRIO addresses")
+        .required(true);
+    parser.addArgument("--verbose")
+        .action(Arguments.storeTrue())
+        .setDefault(false);
+    parser.addArgument("--dynamic_libraries")
+        .type(String.class)
+        .nargs("*")
+        .help("the .so libraries which the executable requires")
+        .setDefault();
+    parser.addArgument("--skip_dynamic_libraries")
+        .action(Arguments.storeTrue())
+        .help(
+            "don't deploy the dynamic libraries (warning: unsafe, might lead to unexpected issues)")
+        .setDefault(false);
+
+    Namespace parsedArgs = parser.parseArgsOrFail(args);
+
+    Runfiles runfiles = Runfiles.create();
+
+    File robotBinary = runfile(runfiles, parsedArgs.get("robot_binary"));
+    String robotCommand = parsedArgs.get("robot_command");
+    ArrayList<String> dynamicLibraryPaths = parsedArgs.get("dynamic_libraries");
+    boolean verbose = parsedArgs.get("verbose");
+
+    if (parsedArgs.get("skip_dynamic_libraries")) {
+      if (verbose) {
+        System.out.println("Skipping dynamic libraries");
+      }
+      dynamicLibraryPaths = new ArrayList<>();
+    }
 
-        // Write new robotCommand
-        runCommand(client, String.format("echo %s > /home/lvuser/robotCommand", robotCommand.replace("{}", robotBinaryDestination)), verbose);
-        runCommand(client, "chmod +x /home/lvuser/robotCommand", verbose);
-        runCommand(client, "chown lvuser:ni /home/lvuser/robotCommand", verbose);
-
-        // Copy dynamic libraries
-        for (String dynamicLibraryPath : dynamicLibraryPaths) {
-            File dynamicLibrary = runfile(runfiles, dynamicLibraryPath);
-            progressBar.setExtraMessage(dynamicLibrary.getName());
-            String dynamicLibraryDestination = String.format("/usr/local/frc/third-party/lib/%s", dynamicLibrary.getName());
-            scp.copy(new FileSystemFile(dynamicLibrary), dynamicLibraryDestination);
-            progressBar.step();
-        }
+    String robotBinaryDestination =
+        String.format("/home/lvuser/%s", robotBinary.getName());
 
-        // Restart robot code
-        runCommand(client, "sync", verbose);
-        runCommand(client, "ldconfig", verbose);
-        runCommand(client, ". /etc/profile.d/natinst-path.sh; /usr/local/frc/bin/frcKillRobot.sh -t -r", verbose);
-    }
+    ProgressBar progressBar = new ProgressBarBuilder()
+                                  .setTaskName("Deploying")
+                                  .setInitialMax(dynamicLibraryPaths.size() + 1)
```

Comment:
In the future, adding a linter / formatter should be a separate PR. This diff is now very complicated to make sense of. I also would have stuck with a 4-space indentation so not every line has to change